expr/shaper.go

```diff
@@ -64,28 +64,27 @@ func createOp(in, out *zng.TypeRecord) (op, error) {
 		switch {
 		case inCol.Type.ID() == outCol.Type.ID():
 			if zng.IsContainerType(inCol.Type) {
-				o.append(op{fromIndex: ind, op: copyContainer})
+				s.append(step{fromIndex: ind, op: copyContainer})
 			} else {
-				o.append(op{fromIndex: ind, op: copyPrimitive})
+				s.append(step{fromIndex: ind, op: copyPrimitive})
 			}
 		case zng.IsRecordType(inCol.Type) && zng.IsRecordType(outCol.Type):
-			step, err := createOp(inCol.Type.(*zng.TypeRecord), outCol.Type.(*zng.TypeRecord))
+			child, err := createStep(inCol.Type.(*zng.TypeRecord), outCol.Type.(*zng.TypeRecord))
 			if err != nil {
-				return op{}, err
+				return step{}, err
 			}
-			step.fromIndex = ind
-			o.append(step)
+			child.fromIndex = ind
+			s.append(child)
 		case zng.IsPrimitiveType(inCol.Type) && zng.IsPrimitiveType(outCol.Type):
-			step := op{fromIndex: ind, op: castPrimitive, castTypes: struct{ from, to zng.Type }{inCol.Type, outCol.Type}}
-			o.append(step)
+			s.append(step{fromIndex: ind, op: castPrimitive, castTypes: struct{ from, to zng.Type }{inCol.Type, outCol.Type}})
 		default:
-			return op{}, fmt.Errorf("createOp incompatible column types %s and %s\n", inCol.Type, outCol.Type)
+			return step{}, fmt.Errorf("createOp incompatible column types %s and %s\n", inCol.Type, outCol.Type)
 		}
 	}
-	return o, nil
+	return s, nil
 }
 
-func (s op) castPrimitive(in zcode.Bytes, b *zcode.Builder) {
+func (s step) castPrimitive(in zcode.Bytes, b *zcode.Builder) {
```

Comment:
By my count, a `step` occupies 13 * 8 bytes, so a pointer receiver seems like a good idea.
```suggestion
func (s *step) castPrimitive(in zcode.Bytes, b *zcode.Builder) {
```

Reply:
👍